### ISSUE TITLE
After upgrade to unreal 5.5 always crash.

### ISSUE THREAD

**Issue by yolohz** [VesCodes/ImGui#32]
appError called: Assertion failed: (ObjectItem->RefCount == 0 && (ObjectItem->GetFlags() & EInternalObjectFlags::RefCounted) == EInternalObjectFlags::None) || GExitPurge [File:C:\UnrealEngine\Engine\Source\Runtime\CoreUObject\Private\UObject\UObjectArray.cpp] [Line: 363] 

any idea why always crash ?

**Comment by TheJamsh** [comment]
The issue seems to be caused by the explicit name passed to UTexture2D::CreateTransient in FImGuiContext::BeginFrame. It seems that the engine tries to recycle an existing object with a non-zero reference count. (Presumably because another FImGuiContext is holding onto one via the TStrongObjectptr.) Not sure what changed in 5.5.

I was only able to hit this in PIE with a Listen Server + Client. Passing either NAME_None or using MakeUniqueObjectName() seems to solve it without any side effects so far.

**Comment by VesCodes** [comment]
Yeah, the name collision is the cause here. I've been waiting to hear back from https://github.com/VesCodes/ImGui/issues/20#issuecomment-2527600654 but have some pending changes in this area.

**Comment by VesCodes** [comment]
Should be addressed with https://github.com/VesCodes/ImGui/commit/e32ab5ab940207538409a8cf79074ed8ab8dfb0f. Let me know if issue persists :)
